ui/app/src/components/CreateDashboardDialog/CreateDashboardDialog.tsx

```diff
@@ -0,0 +1,107 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ChangeEvent, Dispatch, DispatchWithoutAction, useCallback, useState } from 'react';
+import { Dialog, DialogTitle, DialogContent, DialogActions, Button, TextField, IconButton } from '@mui/material';
+import CloseIcon from 'mdi-material-ui/Close';
+
+export interface CreateDashboardDialogProps {
+  open: boolean;
+  onClose: DispatchWithoutAction;
+  onSuccess?: Dispatch<string>;
+}
+
+/**
+ * Dialog used to create a dashboard.
+ * @param props.open Define if the dialog should be opened or not.
+ * @param props.closeDialog Provides the function to close itself.
+ * @param props.onConfirm Action to perform when user confirmed.
+ * @param props.projectName The project where the dashboard will be created.
+ * @constructor
+ */
+const CreateDashboardDialog = (props: CreateDashboardDialogProps) => {
+  const { open, onClose, onSuccess } = props;
+  const [name, setName] = useState<string>();
+  const [error, setError] = useState<string>();
+
+  const handleChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    if (!e.target.value) {
+      setError('Required');
+    } else {
+      setError(undefined);
+    }
+  }, []);
+
+  // Reinitialize form for next time the dialog is opened
+  const resetForm = useCallback(() => {
+    setName(undefined);
+    setError(undefined);
+  }, []);
+
+  const handleSubmit = useCallback(() => {
+    if (name) {
+      onClose();
+      if (onSuccess) {
+        onSuccess(name);
+      }
+      resetForm();
+    }
+  }, [name, onClose, onSuccess, resetForm]);
+
+  const handleClose = useCallback(() => {
+    onClose();
+    resetForm();
+  }, [onClose, resetForm]);
+
+  return (
+    <Dialog open={open} onClose={handleClose} aria-labelledby="confirm-dialog">
+      <DialogTitle id="confirm-dialog">Create Dashboard</DialogTitle>
+      <IconButton
+        aria-label="Close"
+        onClick={handleClose}
+        sx={(theme) => ({
+          position: 'absolute',
+          top: theme.spacing(0.5),
+          right: theme.spacing(0.5),
+        })}
+      >
+        <CloseIcon />
+      </IconButton>
+      <DialogContent dividers sx={{ width: '500px' }}>
+        <TextField
+          required
+          margin="dense"
+          id="name"
+          label="Name"
+          type="text"
+          fullWidth
+          onChange={handleChange}
+          value={name}
+          error={!!error}
+          helperText={error}
+        />
+      </DialogContent>
+      <DialogActions>
+        <Button variant="contained" disabled={!!error} onClick={handleSubmit}>
+          Add
+        </Button>
+        <Button variant="outlined" color="secondary" onClick={handleClose}>
+          Cancel
+        </Button>
+      </DialogActions>
+    </Dialog>
+  );
+};
+
+export default CreateDashboardDialog;
```

Comment:
We should probably add an eslint rule for this, but for consistency we've been trying to avoid default exports in favor of named exports, so the import in ViewProject.tsx would become:
```
import { CreateDashboardDialog } from '../components/CreateDashboardDialog/CreateDashboardDialog';
```